src/urbit/app/chess.hoon

```diff
@@ -766,6 +872,36 @@
                 games  %+  ~(put by games)  u.game-id
                        [game position |2.u.game-state]
               ==
+            %chess-undo-request
+              :-  :~  :*  %give  %fact  ~[/game/(scot %da u.game-id)/updates]
+                          %chess-update  !>([%undo-request u.game-id])
+                      ==
+                  ==
+              %=  this
+                games  (~(put by games) u.game-id u.game-state(got-undo-request &))
+              ==
+            %chess-undo-decline
+              :-  :~  :*  %give  %fact  ~[/game/(scot %da u.game-id)/updates]
+                          %chess-update  !>([%undo-declined u.game-id])
+                      ==
+                  ==
+              %=  this
+                games    (~(put by games) u.game-id u.game-state(sent-undo-request |))
+              ==
+            %chess-undo-accept
+              ?.  sent-undo-request.u.game-state
+                [~ this]
+              :-  :~  :*  %give  %fact  ~[/game/(scot %da u.game-id)/updates]
+                          %chess-update  !>([%undo-accepted u.game-id])
+                      ==
+                  ==
+              =,  game.u.game-state
```

Comment:
This isn't a bad idea, actually, if it were set to `=,  u.game-state`. This could be useful in the `%accept-undo` block above too: rename `game` to `game-state` and add this block after checking that `game-state` is non-null.